src/scheduled_tasks.rs

```diff
@@ -476,23 +485,49 @@ async fn update_instance_software(
           .domain(instance.domain.clone())
           .updated(Some(naive_now()))
           .build();
-        let form = match client.get(&node_info_url).send().await {
+
+        // First, fetch their /.well-known/nodeinfo, then extract the correct nodeinfo link from it
+        let well_known_url = format!("https://{}/.well-known/nodeinfo", instance.domain);
+
+        let form = match client.get(&well_known_url).send().await {
           Ok(res) if res.status().is_client_error() => {
-            // Instance doesn't have nodeinfo but sent a response, consider it alive
+            // Instance doesn't have well-known but sent a response, consider it alive
             Some(default_form)
           }
-          Ok(res) => match res.json::<NodeInfo>().await {
-            Ok(node_info) => {
-              // Instance sent valid nodeinfo, write it to db
-              let software = node_info.software.as_ref();
-              Some(
-                InstanceForm::builder()
-                  .domain(instance.domain)
-                  .updated(Some(naive_now()))
-                  .software(software.and_then(|s| s.name.clone()))
-                  .version(software.and_then(|s| s.version.clone()))
-                  .build(),
-              )
+          Ok(res) => match res.json::<NodeInfoWellKnown>().await {
+            Ok(well_known) => {
+              // Find the first link where the rel contains the allowed rels above
+              match well_known
+                .links
+                .into_iter()
+                .find(|links| allowed_rels.contains(&links.rel))
```

Comment:
This could be optimized so checking if the url starts with `http://nodeinfo.diaspora.software/ns/schema/2.` is only done once

Reply:
Makes sense, thx.